src/Surfnet/StepupGateway/SamlStepupProviderBundle/Provider/ConnectedServiceProviders.php

```diff
@@ -18,67 +18,47 @@
 
 namespace Surfnet\StepupGateway\SamlStepupProviderBundle\Provider;
 
+use Surfnet\SamlBundle\Entity\ServiceProvider;
 use Surfnet\StepupGateway\GatewayBundle\Service\SamlEntityService;
 use Surfnet\StepupGateway\SamlStepupProviderBundle\Exception\InvalidArgumentException;
 use Surfnet\StepupGateway\SamlStepupProviderBundle\Exception\UnknownProviderException;
 
 final class ConnectedServiceProviders
 {
-    private $connected;
+    /**
+     * @var AllowedServiceProviders
+     */
+    private $allowed;
 
     /**
-     * @var \Surfnet\StepupGateway\GatewayBundle\Service\SamlEntityService
+     * @var SamlEntityService
      */
     private $samlEntityService;
 
-    public function __construct(SamlEntityService $samlEntityService, array $connectedServiceProviders)
+    public function __construct(SamlEntityService $samlEntityService, AllowedServiceProviders $allowed)
     {
         $this->samlEntityService = $samlEntityService;
-
-        foreach ($connectedServiceProviders as $serviceProvider) {
-            $this->connectServiceProvider($serviceProvider);
-        }
+        $this->allowed = $allowed;
     }
 
-    /**
-     * @param string $serviceProvider
-     * @return bool
-     */
-    public function isConnected($serviceProvider)
+    public function isConnected(string $serviceProvider): bool
     {
         if (!is_string($serviceProvider)) {
```

Comment:
serviceProvider is passed by as a string, seems that this condition is always false?